reviewrot/basereview.py

```diff
@@ -82,6 +83,62 @@ def check_request_state(self, created_at,
                 raise ValueError("Invalid duration type: %s" % duration)
         return True
 
+    def _decode_response(self,response):
+
+        GERRIT_JSON_PREFIX = ")]}\'\n"
+
+        """ Remove Gerrit's prefix and convert to JSON.
+        :returns:
+            Converted JSON content
+        :raises:
+            requests.HTTPError if the response contains an HTTP error status code.
+        """
+        content = response.content.strip()
+        if response.encoding:
+            content = content.decode(response.encoding)
+        response.raise_for_status()
+        content_type = response.headers.get('content-type', '')
+        if content_type.split(';')[0] != 'application/json':
+            return content
+        if content.startswith(GERRIT_JSON_PREFIX):
+            content = content[len(GERRIT_JSON_PREFIX):]
+        try:
+            return json.loads(content)
+        except ValueError:
+            logging.error('Invalid json content: %s' % content)
+            raise
+
+
+    def _call_api(self, url, method='GET', ssl_verify=True):
+        """
+        Method used to call the API.
+        It returns the raw JSON returned by the API or raises an exception
+        if something goes wrong.
+        Args:
+            method (str): the URL to call, can be GET, POST, DELETE, UPDATE...
+                          Defaults to GET
+        Returns:
+            raw JSON returned by API
+        """
+        response = self.session.request(method=method,url=url,headers=self.header,verify=ssl_verify)
+        decoded_response = None
+        try:
+
+            decoded_response = response.json()
+        except Exception as err:
+            try:
+                decoded_response = self._decode_response(response)
+            except HTTPError as http_error:
+                return ''
```

Comment:
I'd suggest adding something like: 
log.exception("error calling API at %s" % url)
Above the above return. That way it's going to be obvious when something went wrong